.github/workflows/main_workflow.yml

```diff
@@ -145,46 +142,123 @@ jobs:
 
     strategy:
       matrix:
-        os: [ ubuntu-latest ] # TODO add [macos-latest, windows-latest]
-        python-version: [ "3.10" ] # TODO add [3.8, 3.9]
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+        python-version: [ "3.7", "3.8", "3.9", "3.10" ]
 
     timeout-minutes: 30
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - uses: actions/setup-python@v3
         with:
           submodules: recursive
           python-version: ${{ matrix.python-version }}
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "1.17.7"
       - if: matrix.os == 'ubuntu-latest'
         name: Install dependencies (ubuntu-latest)
         run: |
           sudo apt-get update --fix-missing
           sudo apt-get autoremove
           sudo apt-get autoclean
-          pip install tox
+          pip install tomte[tox]==0.1.5
           pip install --upgrade setuptools==60.10.0
+
           # install Protobuf compiler
           wget https://github.com/protocolbuffers/protobuf/releases/download/v3.11.4/protoc-3.11.4-linux-x86_64.zip
           unzip protoc-3.11.4-linux-x86_64.zip -d protoc
           sudo mv protoc/bin/protoc /usr/local/bin/protoc
+
           # install IPFS
           sudo apt-get install -y wget
           wget -O ./go-ipfs.tar.gz https://dist.ipfs.io/go-ipfs/v0.6.0/go-ipfs_v0.6.0_linux-amd64.tar.gz
           tar xvfz go-ipfs.tar.gz
           sudo mv go-ipfs/ipfs /usr/local/bin/ipfs
           ipfs init
+
           # install Tendermint
           wget -O tendermint.tar.gz https://github.com/tendermint/tendermint/releases/download/v0.34.19/tendermint_0.34.19_linux_amd64.tar.gz
           tar -xf tendermint.tar.gz
           sudo mv tendermint /usr/local/bin/tendermint
+
           # pull node image needed for the integration tests
+          docker pull node:16.7.0
           docker pull trufflesuite/ganache:beta
-          docker pull valory/autonolas-registries:latest
```

Comment:
This is required by default

Reply:
All this issues you correctly pointed come from my last commit, where I tried to fix the worflow error overwriting the current from the one in the autonomous fund. Reverted now.